# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/LocalAccessible.cpp
# Commit: 8d31c0604efe
# Full Hash: 8d31c0604efef2015995df32766d38dd4dd45ca2
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-08-04 04:34:13
# Regressor Bug: 1781536
# File Overlap Count: 1
# Description:
#   Bug 1781536 part 1: Support base Accessible in Relation. r=morgan
#   
#   Previously, Relation::Next returned a LocalAccessible.
#   Now, it returns an Accessible so that it can support RemoteAccessible as well.
#   Relation::LocalNext has been added as a convenience to return a LocalAccessible when the caller is certain that only LocalAccessible is relevant.
# ==============================================================================

diff -r af7f9589f48c -r 8d31c0604efe accessible/generic/LocalAccessible.cpp
--- a/accessible/generic/LocalAccessible.cpp	Thu Aug 04 02:47:34 2022 +0300
+++ b/accessible/generic/LocalAccessible.cpp	Wed Aug 03 23:58:50 2022 +0000
@@ -1480,7 +1480,7 @@
         // If focus is in a child of the tab panel surely the tab is selected!
         Relation rel = RelationByType(RelationType::LABEL_FOR);
         LocalAccessible* relTarget = nullptr;
-        while ((relTarget = rel.Next())) {
+        while ((relTarget = rel.LocalNext())) {
           if (relTarget->Role() == roles::PROPERTYPAGE &&
               FocusMgr()->IsFocusWithin(relTarget)) {
             state |= states::SELECTED;
@@ -1766,7 +1766,7 @@
       // Listbox is owned by a combobox
       Relation rel = RelationByType(RelationType::NODE_CHILD_OF);
       LocalAccessible* targetAcc = nullptr;
-      while ((targetAcc = rel.Next())) {
+      while ((targetAcc = rel.LocalNext())) {
         if (targetAcc->IsCombobox()) return roles::COMBOBOX_LIST;
       }
     }
@@ -2470,14 +2470,14 @@
   // a name/description provider is added to doc.
   Relation rel = RelationByType(RelationType::LABELLED_BY);
   LocalAccessible* relTarget = nullptr;
-  while ((relTarget = rel.Next())) {
+  while ((relTarget = rel.LocalNext())) {
     if (!relTarget->HasNameDependent()) {
       relTarget->ModifySubtreeContextFlags(eHasNameDependent, true);
     }
   }
 
   rel = RelationByType(RelationType::DESCRIBED_BY);
-  while ((relTarget = rel.Next())) {
+  while ((relTarget = rel.LocalNext())) {
     if (!relTarget->HasDescriptionDependent()) {
       relTarget->ModifySubtreeContextFlags(eHasDescriptionDependent, true);
     }
@@ -3591,7 +3591,7 @@
         rel.AppendIter(new IDRefsIterator(mDoc, mContent, relAtom));
       }
 
-      while (LocalAccessible* acc = rel.Next()) {
+      while (LocalAccessible* acc = rel.LocalNext()) {
         ids.AppendElement(acc->IsDoc() ? 0 : acc->ID());
       }
       if (ids.Length()) {